File: AtlasWeb/Core/Modules/Tibbi_Surec/Ketem_Modulu/CigaretteInspectionFormViewModel.ts
//$473C0608
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { CigaretteExamination } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class CigaretteInspectionFormViewModel extends BaseViewModel {
    @Type(() => CigaretteExamination)
    public _CigaretteExamination: CigaretteExamination = new CigaretteExamination();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Ketem_Modulu/CigaretteInspectionFormViewModel.cs
//$B6F49DAF
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Microsoft.AspNetCore.Mvc;
using System.ComponentModel;

namespace Core.Controllers
{
    public partial class CigaretteExaminationServiceController
    {
        partial void PreScript_CigaretteInspectionForm(CigaretteInspectionFormViewModel viewModel, CigaretteExamination cigaretteExamination, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._CigaretteExamination.SubEpisode = episodeAction.SubEpisode;

            }

            ContextToViewModel(viewModel, objectContext);
        }

        
    }
}

namespace Core.Models
{
    public partial class CigaretteInspectionFormViewModel: BaseViewModel
    {
    }
}
